### DISCUSSION TITLE
Is it possible to use Storybook 8?

### DISCUSSION THREAD

**Discussion by djMax** [storybookjs/react-native#575] [Q&A] [answered]
I have a use case that will reside in the same repo as a web component library that uses Storybook 8. I tried putting RN storybook in a subdirectory, but I'm unable to "bust out" of that subdirectory for imports for some reason. BUT, is it even possible to put storybook/react-native on top of a storybook 8 set of core modules?

**Comment by dannyhw** [comment]
@djMax storybook 8 for react native is still in alpha

**Comment by djMax** [comment]
Any way for me to get a look at that? I can't back off on sb8 web because there's just too much infra, but the rn bits are from 0, so I could definitely start simple there.

**Reply by dannyhw** [reply]
Its here
https://github.com/storybookjs/react-native/pull/538

**Comment by trajano** [comment]
Just attempted to do an upgrade. major failure though but then I have a monorepo running expo so it may not have been a tested configuration.  Right now I am getting 
> Unable to resolve "@storybook/core/manager-api" from "..\..\node_modules\@storybook\react-native\dist\index.js"

Which doesn't really make sense yet since I thought it was "@storybook/manager-api"

**Reply by trajano** [reply]
Seems to be because the metro.config.ts isn't supported yet https://github.com/facebook/metro/issues/916 so now it loads up next bit of issue is the backgrounds stopped working

**Reply by trajano** [reply]
Eventually got things working.
https://github.com/trajano/expo-experiments/commit/bf5d881ee75e7260b5cb37eacb2dda55ef0264cf

I haven't found out where the "storybook" branding on the menu is set though.

**Reply by dannyhw** [reply]
Ive been pushing some fixes around the metro stuff, make sure to update to the latest version. Also many packages were consolidated under storybook/core in 8.3.

**Reply by dannyhw** [reply]
Only theming is possible currently , are you looking to replace the logo/name? Open to suggestions/prs for that.

**Reply by trajano** [reply]
The `Theme` object has `brand.title` but doesn't seem to work.  As far as metro and TS support feel free to gank [withStoryBook.d.ts](https://github.com/trajano/expo-experiments/commit/bf5d881ee75e7260b5cb37eacb2dda55ef0264cf#diff-59f1ef07bc38e05888c99a3c5af519543288158a517bbb7973b42446f354812b) to provide the necessary typings.

**Reply by dannyhw** [reply]
the theme is just copied from the web theme so thats something I missed when implementing I guess

**Reply by trajano** [reply]
I just checked that the withStorybook.d.ts just needs to remove the declare module declaration.  I just created a [patch for myself](https://github.com/trajano/expo-experiments/blob/629a7efa36cc7910ceffc53536e3f1ef809e2187/patches/%40storybook%2Breact-native%2B8.3.9.patch#L24-L96) which incorporates your documentation in the README.

**Reply by trajano** [reply]
> Only theming is possible currently , are you looking to replace the logo/name? Open to suggestions/prs for that.

The Logo and DarkLogo are embedded in the code here

https://github.com/storybookjs/react-native/blob/6f62dd342dd4eeba6eda94c3deb88029599d5990/packages/react-native-ui/src/Layout.tsx#L77C1-L81C19

and

https://github.com/storybookjs/react-native/blob/6f62dd342dd4eeba6eda94c3deb88029599d5990/packages/react-native-ui/src/Layout.tsx#L140-L144

The logo includes the `storybook` text.  To get around this that segment of the code needs to read from `brand` and assemble as needed.

I think the way to handle this is to extract a new component to provide the Brand data something along the lines of

```tsx
const BrandLogo = (theme: ThemeVars) => {
  const openBrandUrl = useCallback(..., [theme]);
  const brandTitleStyle= useMemo(()=> {
      return [ styles.brandTitle, { color: theme.??? } ]
    }, [theme]);
  const imageSource =  useMemo<ImageSourcePropType|undefined>(()=> {
     if (theme.imageSource) { return theme.imageSource }
     else if (theme.image) { return { uri: image } }
     else return undefined;
    }, [theme]);
  if (!theme.brand) {
    return {theme.base === 'light' ? (
                  <Logo height={25} width={125} />
                ) : (
                  <DarkLogo height={25} width={125} />
                )}
  } else {
    return  <Pressable onPress={openBrandUrl}>
         <View style={styles.brandLogo}>
         <Image source={imageSource}> /// how to handle if image is missing
          <Text style={brandTitleStyle}>{theme.brandTitle}</Text> /// how to handle if text is missing
      </View>
  }
}
const styles = StyleSheet.create({
  brandLogo: {
    width: 125,
    height: 25,
    flexDirection: row
  }, brandTitle:{
???
  }
})
```

In addition `brand.image` needs to support something aside from `string|null` but rather ImageSourcePropType which means `image` cannot be reasonably used, so the `brand` typing also needs to be augmented to add `imageSource: ImageSourcePropType` and use that.  image would be

**Comment by dannyhw** [comment, marked as answer]
It is now possible, please check the migration docs https://github.com/storybookjs/react-native/blob/next/MIGRATION.md#from-version-76x-to-83x
